src/app/doc_exporter.cpp

```diff
@@ -390,25 +390,23 @@ class DocExporter::SimpleLayoutSamples : public DocExporter::LayoutSamples {
         return;
       token.set_progress(0.2f + 0.2f * i / samples.size());
 
-      if (sample.isLinked()) {
-        ++i;
-        continue;
-      }
-
       if (sample.isEmpty()) {
         sample.setInTextureBounds(gfx::Rect(0, 0, 0, 0));
         ++i;
         continue;
       }
 
-      if (m_mergeDups) {
+      if (m_mergeDups || sample.isLinked()) {
         doc::ImageBufferPtr sampleBuf = std::make_shared<doc::ImageBuffer>();
         doc::ImageRef sampleRender(sample.createRender(sampleBuf));
         auto it = duplicates.find(sampleRender);
         if (it != duplicates.end()) {
           const uint32_t j = it->second;
 
-          sample.setDuplicated();
+          if (sample.isLinked())
+            sample.setLinked();
```

Comment:
This statement does nothing (as if the `sample` is already linked).

It would be nice if we can add some test cases in the `tests` repository (similar to the shared tests, something that is failing now and this PR fixes).